src/test/java/com/fasterxml/jackson/databind/deser/jdk/EnumDeserializationTest.java

```diff
@@ -527,4 +530,33 @@ public void testExceptionFromCustomEnumKeyDeserializer() throws Exception {
             assertTrue(e.getMessage().contains("Undefined AnEnum"));
         }
     }
+
+    // see here for real world example:
+    // https://github.com/facebook/facebook-java-business-sdk/blob/v3.2.11/src/main/java/com/facebook/ads/sdk/CustomAudience.java#L4139
+    static enum EnumWithToStringNullValue {
+        NON_NULL("NON_NULL"),
+        NULL(null);
+
+        private String value;
+
+        private EnumWithToStringNullValue(String value) {
+            this.value = value;
+        }
+
+        @Override
+        public String toString() {
+            return value;
+        }
+    }
+
+    static class PojoWithEnum {
+      public EnumWithToStringNullValue enumWithToStringNullValue;
+    }
+
+    public void testEnumWithToStringNullValue() {
+        ObjectMapper mapper = new ObjectMapper().enable(READ_ENUMS_USING_TO_STRING);
+        ObjectNode on = mapper.createObjectNode();
+        on.put("enumWithToStringNullValue", "NON_NULL");
```

Comment:
w/out the supplied "fix", this throws an exception even through I'm not mapping the NULL enum.